### PR TITLE
Added "Find Text in PDF" action

### PR DESCRIPTION
**Pull request by ionutmarisca**
Signed-off by: Ionut Mihai Marisca <ionut-mihai.marisca@hpe.com>

### CODE REVIEW

File: cs-utilities/src/main/java/io/cloudslang/content/utilities/actions/FindTextInPdf.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+ * Created by marisca on 7/11/2017.
+ */
+public class FindTextInPdf {
+
+    /**
+     * This operation checks if a text input is found in a PDF file.
+     *
+     * @param text       The text to be searched for in the PDF file.
+     * @param ignoreCase Whether to ignore if characters of the text are lowercase or uppercase.
+     *                   Valid values: "true", "false". For any other value the ignoreCase will be set to "false".
```
Comment (victorursan) [inline cs-utilities/src/main/java/io/cloudslang/content/utilities/actions/FindTextInPdf.java]: usually only "true" or "false" are allowed, other values are considered and error ( invalid user input, think about  "si" or "\t true" will both be considered as "false" an the user will not know why the behavior is not the expected one)

File: cs-utilities/src/main/java/io/cloudslang/content/utilities/actions/FindTextInPdf.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            @Param(value = TEXT, required = true, description = INITIAL_VALUE_DESC) String text,
+            @Param(value = IGNORE_CASE, description = IGNORE_CASE_DESC) String ignoreCase,
+            @Param(value = PATH_TO_FILE, required = true, description = DEFAULT_VALUE_DESC) String pathToFile,
+            @Param(value = PASSWORD, description = PASSWORD_DESC, encrypted = true) String password) {
+
+        try {
+            final Path path = Paths.get(pathToFile);
+            final String pdfPassword = defaultIfEmpty(password, "");
+            final String pdfContent = PdfParseService.getPdfContent(path, pdfPassword).trim().replace(System.lineSeparator(), "");
+            final boolean validIgnoreCase = toBoolean(defaultIfEmpty(ignoreCase, FALSE));
```
Comment (victorursan) [inline cs-utilities/src/main/java/io/cloudslang/content/utilities/actions/FindTextInPdf.java]: you can use cs-commons -> [BooleanUtils.toBoolean](https://github.com/CloudSlang/cs-actions/blob/5fe9ee53cb58ad88d7dabc82b0867e808ef842bd/cs-commons/src/main/java/io/cloudslang/content/utils/BooleanUtilities.java#L75) but before, you should validate it's "true" or "false" [BooleanUtils.isValid](https://github.com/CloudSlang/cs-actions/blob/5fe9ee53cb58ad88d7dabc82b0867e808ef842bd/cs-commons/src/main/java/io/cloudslang/content/utils/BooleanUtilities.java#L45)

File: cs-utilities/src/main/java/io/cloudslang/content/utilities/actions/FindTextInPdf.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+            if (validIgnoreCase)
+                return getSuccessResultsMap(valueOf(countMatches(pdfContent.toLowerCase(), text.toLowerCase())));
+
+            return getSuccessResultsMap(valueOf(countMatches(pdfContent, text)));
```
Comment (victorursan) [inline cs-utilities/src/main/java/io/cloudslang/content/utilities/actions/FindTextInPdf.java]: possible duplication of code? `valueOf(countMatches(pdfContent...., text....)` maybe extract the if in a method with 3 params `(pdfContent, text, validIgnoreCase)` and call `getSuccessResultsMap` on the result

File: cs-utilities/src/main/java/io/cloudslang/content/utilities/entities/constants/Inputs.java [hunk 4]
```diff
@@ -17,5 +18,8 @@
     public static final String INITIAL_VALUE = "initialValue";
     public static final String DEFAULT_VALUE = "defaultValue";
     public static final String TRIM = "trim";
-
+    public static final String TEXT = "text";
+    public static final String IGNORE_CASE = "ignoreCase";
+    public static final String PATH_TO_FILE = "pathToFile";
+    public static final String PASSWORD = "password";
```
Comment (victorursan) [inline cs-utilities/src/main/java/io/cloudslang/content/utilities/entities/constants/Inputs.java]: @robert-sandor will like this :)) #STAT

File: cs-utilities/src/main/java/io/cloudslang/content/utilities/services/PdfParseService.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        final PDDocument document;
+
+        if (password.isEmpty()) {
+            document = PDDocument.load(path.toFile());
+        } else {
+            document = PDDocument.load(path.toFile(), password);
+        }
+
+        final String pdfContent = new PDFTextStripper().getText(document);
+        document.close();
```
Comment (victorursan) [inline cs-utilities/src/main/java/io/cloudslang/content/utilities/services/PdfParseService.java]: [IDK what documentation](https://pdfbox.apache.org/docs/2.0.2/javadocs/org/apache/pdfbox/pdmodel/PDDocument.html) says PDDocument document is AutoCloseable, so look at it and maybe use try with resources

File: cs-utilities/src/main/java/io/cloudslang/content/utilities/services/PdfParseService.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+import java.io.IOException;
+import java.nio.file.Path;
+
+/**
+ * Created by marisca on 7/11/2017.
+ */
+public class PdfParseService {
+
+    public static String getPdfContent(final Path path, final String password) throws IOException {
+        final PDDocument document;
```
Comment (victorursan) [inline cs-utilities/src/main/java/io/cloudslang/content/utilities/services/PdfParseService.java]: maybe create a method that returns the document ( so you don't define it as empty and assign the value afterwards)

File: cs-utilities/src/test/java/io/cloudslang/content/utilities/FindTextInPdfTest.java [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+/**
+ * Created by marisca on 7/11/2017.
+ */
+public class FindTextInPdfTest {
+    private final FindTextInPdf findTextInPdf = new FindTextInPdf();
+
+    @Test
+    public void samplePdfSuccessfulTest() {
+        try {
+            URL resource = FindTextInPdf.class.getClassLoader().getResource("pdf/sample-pdf-1.pdf");
```
Comment (victorursan) [inline cs-utilities/src/test/java/io/cloudslang/content/utilities/FindTextInPdfTest.java]: no more 
``` java
final
``` 
? 😢

File: cs-utilities/src/test/java/io/cloudslang/content/utilities/FindTextInPdfTest.java [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+
+/**
+ * Created by marisca on 7/11/2017.
+ */
+public class FindTextInPdfTest {
+    private final FindTextInPdf findTextInPdf = new FindTextInPdf();
+
+    @Test
+    public void samplePdfSuccessfulTest() {
+        try {
```
Comment (victorursan) [inline cs-utilities/src/test/java/io/cloudslang/content/utilities/FindTextInPdfTest.java]: why try { } catch ? ... that catch will hide fails 😆  ... add it to the method signature

File: cs-utilities/src/test/java/io/cloudslang/content/utilities/FindTextInPdfTest.java [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+                assertEquals("1", result.get(RETURN_RESULT));
+            }
+        } catch (URISyntaxException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void samplePdfSuccessfulTestIgnoreCase() {
+        try {
```
Comment (victorursan) [inline cs-utilities/src/test/java/io/cloudslang/content/utilities/FindTextInPdfTest.java]: same here and bellow, there is an internal repo cs-documentation (or something like this with best practices that state that you shouldn't use try { } catch { } in some tests ) @tethryus  can show you the repo

File: cs-utilities/src/test/java/io/cloudslang/content/utilities/FindTextInPdfTest.java [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+    }
+
+    @Test
+    public void samplePdfFailureTest() {
+        try {
+            URL resource = FindTextInPdf.class.getClassLoader().getResource("pdf/sample-pdf-1.pdf");
+            if (resource != null) {
+                File file = new File(resource.toURI());
+                final Map<String, String> result = findTextInPdf.execute("This", "false", file.toString() + ".wrong", "");
+                System.out.println(result);
```
Comment (victorursan) [inline cs-utilities/src/test/java/io/cloudslang/content/utilities/FindTextInPdfTest.java]: ? `System.out.println` you rebel 😆

File: cs-utilities/pom.xml [hunk 11]
```diff
@@ -60,16 +68,18 @@
...
         <!--Dependencies versions-->
         <score-content-sdk.version>1.10.6</score-content-sdk.version>
         <cs-commons.version>0.0.4</cs-commons.version>
         <junit.version>4.12</junit.version>
+        <pdfbox.version>2.0.8</pdfbox.version>
         <!--Misc properties-->
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <maven.compiler.source>1.7</maven.compiler.source>
         <maven.compiler.target>1.7</maven.compiler.target>
+        <custom.nexus>overwriten_by_ci</custom.nexus>
```
Comment (victorursan) [inline cs-utilities/pom.xml]: is this overwritten in some cases ?

File: cs-utilities/pom.xml [hunk 12]
```diff
@@ -60,16 +68,18 @@
...
+        <maven-deploy-plugin.version>2.8.2</maven-deploy-plugin.version>
         <maven-gpg-plugin.version>1.5</maven-gpg-plugin.version>
         <!--Dependencies versions-->
         <score-content-sdk.version>1.10.6</score-content-sdk.version>
         <cs-commons.version>0.0.4</cs-commons.version>
         <junit.version>4.12</junit.version>
+        <pdfbox.version>2.0.8</pdfbox.version>
         <!--Misc properties-->
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <maven.compiler.source>1.7</maven.compiler.source>
```
Comment (victorursan) [inline cs-utilities/pom.xml]: 😢

**Comment by victorursan** [comment]
adding Shuni @robert-sandor

File: cs-utilities/src/main/java/io/cloudslang/content/utilities/services/PdfParseService.java [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+
+        if (password.isEmpty()) {
```
Comment (robert-sandor) [inline cs-utilities/src/main/java/io/cloudslang/content/utilities/services/PdfParseService.java]: Also, why not use `StringUtils.isEmpty()` here ?

File: cs-utilities/src/main/java/io/cloudslang/content/utilities/actions/FindTextInPdf.java [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+            })
+    public Map<String, String> execute(
+            @Param(value = TEXT, required = true, description = INITIAL_VALUE_DESC) String text,
+            @Param(value = IGNORE_CASE, description = IGNORE_CASE_DESC) String ignoreCase,
+            @Param(value = PATH_TO_FILE, required = true, description = DEFAULT_VALUE_DESC) String pathToFile,
+            @Param(value = PASSWORD, description = PASSWORD_DESC, encrypted = true) String password) {
+
+        try {
+            final Path path = Paths.get(pathToFile);
+            final String pdfPassword = defaultIfEmpty(password, "");
```
Comment (robert-sandor) [inline cs-utilities/src/main/java/io/cloudslang/content/utilities/actions/FindTextInPdf.java]: Also here, you could use `StringUtils.EMPTY` instead

File: cs-utilities/pom.xml (hunk 11 above)
Comment (tethryus) [reply cs-utilities/pom.xml]: It will be overwritten by the Jenkins job.

**Comment by robert-sandor** [comment]
@victorursan are the changes accepted ?

**Pull request merged**